source/solvers/vof_algebraic_interface_reinitialization.cc

```diff
@@ -0,0 +1,851 @@
+              const Tensor<1, dim> reinitialized_phase_gradient =
+                present_reinitialized_phase_gradient_values[q];
+
+              // Compute normal vector with projected phase gradient
+              const double projected_vof_phase_gradient_norm =
+                projected_vof_phase_gradient.norm() + tolerance;
+              const Tensor<1, dim> interface_normal =
+                (projected_vof_phase_gradient /
+                 projected_vof_phase_gradient_norm);
+
+              // Local rhs assembly
+              for (unsigned int i = 0; i < n_dofs_per_cell; ++i)
+                {
+                  // Time-stepping term (bdf1)
+                  for (unsigned int p = 0; p < 2; ++p)
+                    {
+                      local_rhs(i) -= bdf_coefficient_vector[p] *
+                                      (phase_fraction_values[p] * phi[i]) *
+                                      JxW_vec[q];
+                    }
+
+                  local_rhs(i) -=
+                    (
+                      // Compressive term
+                      phi[i] *
+                        (scalar_product(reinitialized_phase_gradient,
+                                        interface_normal) -
+                         2 * phase_fraction *
+                           scalar_product(reinitialized_phase_gradient,
+                                          interface_normal) -
+                         phase_fraction * (1 - phase_fraction) * curvature) +
+                      // Diffusive term
+                      (scalar_product(
+                        grad_phi[i],
+                        (diffusivity_coefficient *
+                         scalar_product(reinitialized_phase_gradient,
+                                        interface_normal) *
+                         interface_normal)))) *
+                    JxW_vec[q];
+                }
+            }
+          // Distribute the local contributions to the global system
+          cell->get_dof_indices(local_dof_indices);
+          this->zero_constraints.distribute_local_to_global(local_rhs,
+                                                            local_dof_indices,
+                                                            this->system_rhs);
+        }
+    }
+  this->system_rhs.compress(VectorOperation::add);
+}
+
+
+template <int dim>
+void
+VOFAlgebraicInterfaceReinitialization<dim>::solve_linear_system(
+  const bool initial_step,
+  const bool /*renewed_matrix*/)
+{
+  auto mpi_communicator = this->triangulation->get_communicator();
+
+  const AffineConstraints<double> &constraints_used =
+    initial_step ? this->nonzero_constraints : this->zero_constraints;
+
+  const bool verbose(
+    this->subequation_verbosity != Parameters::Verbosity::quiet &&
+    this->simulation_parameters.vof_subequations_linear_solvers
+        .at(VOFSubequationsID::algebraic_interface_reinitialization)
+        .verbosity != Parameters::Verbosity::quiet);
```

Comment:
I don't understand why you check the member of the vof_reinitialization class and the one for the linear solver parameter?